UnityProject/Assets/Scripts/Objects/Engineering/PowerGenerator.cs

```diff
@@ -138,55 +142,59 @@ public bool WillInteract(HandApply interaction, NetworkSide side)
 		{
 			if (!DefaultWillInteract.Default(interaction, side)) return false;
 			if (interaction.TargetObject != gameObject) return false;
-			if (interaction.HandObject != null && !Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.SolidPlasma)) return false;
-
-			return true;
+			if (interaction.HandObject == null) return true;
```

Comment:
Are you sure you want to be checking for null here? As the server interaction is checking the contents of the hand.

Reply:
oh wait sorry didnt check the line below :p

Reply:
So, do I need to change this check?